src/sentry/shared_integrations/client/proxy.py

```diff
@@ -0,0 +1,70 @@
+import time
+
+from django.conf import settings
+from requests import PreparedRequest, Request
+
+from sentry.integrations.client import ApiClient
+from sentry.middleware.integrations.integration_proxy import (
+    PROXY_ADDRESS,
+    PROXY_OI_HEADER,
+    PROXY_SIGNATURE_HEADER,
+    PROXY_TIMESTAMP_HEADER,
+)
+from sentry.silo.base import SiloMode
+from sentry.silo.util import encode_subnet_signature
+
+
+class IntegrationProxyClient(ApiClient):
+    """
+    Universal Client to access third-party resources safely in Hybrid Cloud.
+    Requests to third parties must always exit the Sentry subnet via the Control Silo, and only
+    add sensitive credentials at that stage.
+    """
+
+    should_proxy = False
+
+    def __init__(self, org_integration_id: int) -> None:
+        super().__init__()
+        self.org_integration_id = org_integration_id
+
+        is_region_silo = SiloMode.get_current_mode() == SiloMode.REGION
+        subnet_secret = getattr(settings, "SENTRY_SUBNET_SECRET")
+        control_address = getattr(settings, "SENTRY_CONTROL_ADDRESS")
+
+        if is_region_silo and subnet_secret and control_address:
+            self.should_proxy = True
+            self.proxy_url = f"{settings.SENTRY_CONTROL_ADDRESS}{PROXY_ADDRESS}"
```

Comment:
Should `PROXY_ADDRESS` be `PROXY_PATH`? Joining two addresses together is a bit odd.